cmd/stellar-rpc/internal/ingest/backfill.go

```diff
@@ -0,0 +1,278 @@
+package ingest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stellar/go-stellar-sdk/ingest/ledgerbackend"
+	"github.com/stellar/go-stellar-sdk/support/datastore"
+	supportlog "github.com/stellar/go-stellar-sdk/support/log"
+	"github.com/stellar/go-stellar-sdk/xdr"
+
+	"github.com/stellar/stellar-rpc/cmd/stellar-rpc/internal/config"
+	"github.com/stellar/stellar-rpc/cmd/stellar-rpc/internal/db"
+)
+
+const (
+	// OneDayOfLedgers is (roughly) a 24 hour window of ledgers.
+	OneDayOfLedgers   = config.OneDayOfLedgers
+	SevenDayOfLedgers = config.OneDayOfLedgers * 7
+	// Number of ledgers to read/write at a time during backfill
+	ChunkSize uint32 = OneDayOfLedgers / 4 // 6 hours. Takes X minutes to process
+)
+
+// This function backfills the local database with n ledgers from the datastore
+// It is called by daemon.go if cfg.Backfill is true
+func RunBackfill(cfg *config.Config, logger *supportlog.Entry, localDbConn *db.DB, localDbRW db.ReadWriter, dsInfo DatastoreInfo) error {
+	logger.Infof("Beginning backfill process")
+	var (
+		ctx              context.Context = context.Background()
```

Comment:
We should probably have a context with a reasonably excessive timeout for this process based on the ledger quantity.